homeassistant/components/zha/climate.py

```diff
@@ -207,6 +233,10 @@ def extra_state_attributes(self):
             data[ATTR_PI_HEATING_DEMAND] = self._thrm.pi_heating_demand
         if self._thrm.pi_cooling_demand is not None:
             data[ATTR_PI_COOLING_DEMAND] = self._thrm.pi_cooling_demand
+        if self._thrm.window_open is not None:
+            data[ATTR_WINDOW_OPEN] = WINDOW_DETECTION_2_STATE.get(
```

Comment:
Measurements that are not related to the hvac operation that we control should be separate sensor entities.

Reply:
Thanks for reviewing the PR. I put it there because the etrv stops heating if the window is detected as open (and also because it is reported in the Thermostat cluster). So in that sense I thought it was related to the hvac operation in the same way occupancy is (which is reported as an attribute). Or why should this be different than occupancy?

Reply:
We should only include attributes that represent measurements that can also be can controlled from the device. If it's just a measurement it should be a separate sensor entity.

Old code doesn't always follow our design rules, so it doesn't help looking at that.

Reply:
You can manually set whether a window is open or closed, and it will stop heating in that case (this is also not currently exposed in the UI). Do you think it still needs a separate sensor entity, even if this can be controlled from the device?

Reply:
Zha currently can't do multiple entities based on the single ZCL cluster (thermostat) it's on Todo but no ETA

Reply:
If we name the attribute like 'thermostat_auto_off_on_window_status' would that qualify as a thermostat measurement?

Reply:
It's not about the name. This measurement is not a representation of an operation mode that is controllable (like eg fan mode) and not a measurement of an uncontrollable operation feature (like eg hvac_action). So it should be a separate sensor entity. If we would have implemented the climate integration today, current humidity and current temperature would not have been included as built in attributes either.